aspnet/fundamentals/hosting.rst

```diff
@@ -0,0 +1,298 @@
+
+What is the difference between a host and a server?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The host is responsible for ensuring the application's services and the server are available and properly configured. You can think of the host as being a wrapper around the server. The host is configured to use a particular server; the server is unaware of its host.
+
+Setting up a Host
+-----------------
+
+You create a host using an instance of ``WebHostBuilder``. This is typically done in your app's entry point, which by default will be located in a *Program.cs* file. A typical *Program.cs*, shown below, demonstrates how to use a ``WebHostBuilder`` to build a host. 
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/Program.cs
+  :emphasize-lines: 14-21
+  :language: c#
+  :caption: Program.cs
+
+The ``WebHostBuilder`` is responsible for creating the host that will bootstrap the server for the app. ``WebHostBuilder`` supports many optional extensions, but the specification of an ``IServer`` is required. The built-in server is Kestrel. Use ``UseKestrel`` (with options, if desired) to set it up as your app's server.
+
+The server's *content root* determines where it searches for content files, like MVC View files. The default content root is the folder from which the application is run. Specifying ``Directory.GetCurrentDirectory`` as the content root will use the web project's root folder as the app's content root when the app is started from this folder (for example, calling ``dotnet run`` from the web project folder).
+
+If the app should work with IIS, the ``UseIISIntegration`` method should be called as part of building the host. Note that this does not configure a *server*, like ``UseKestrel`` does. To use IIS with ASP.NET Core, you must specify both ``UseKestrel`` and ``UseIISIntegration``.
+
+.. note:: ``UseKestrel`` and ``UseIISIntegration`` are very different actions. IIS is only used as a reverse proxy. ``UseKestrel`` creates the server the app should use, running on a specified port, and it hosts the code. It registers Kestrel as an ``IServerFactory``. At runtime, it gets the actual ``IServer`` service, instantiates it, and runs it. ``UseIISIntegration`` will look at environment variables used by IIS/IISExpress and will make decisions like which dynamic port to run on, which headers to set, etc. However, it doesn't deal with or create an ``IServer``.
```

Comment:
Too much irrelevant detail.

+.. note:: ``UseKestrel`` and ``UseIISIntegration`` are very different actions. IIS is only used as a reverse proxy. ``UseKestrel`` creates the web server and  hosts the code. It registers Kestrel as an ``IServerFactory``. ``UseIISIntegration`` specifies IIS as the reverse proxy server. It also examines environment variables used by IIS/IISExpress, makes decisions like which dynamic port to use, which headers to set, etc. However, it doesn't deal with or create an ``IServer``.